subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/GradleUserHomeScopeFileTimeStampInspector.java

```diff
@@ -84,4 +88,18 @@ public void beforeComplete() {
             }
         }
     }
+
+    /**
+     * Detects whether the file system has a reliable high precision timestamp.
+     *
+     * In case both timestamp are in millisecond precision,
+     * then the provided file system is definitely reliable, we don't have to discard file hashes, and we will correctly detect file changes.
+     */
+    private boolean isReliableTimestampPrecision(long timestamp) {
+        return isBaseTimestampHighPrecision && isHighTimestampPrecision(timestamp);
```

Comment:
```suggestion
        return isBaseTimestampHighPrecision || isHighTimestampPrecision(timestamp);
```

I thought it is enough that one of the two timestamps is high precision. Then the higher precision one would have detected the change, right?

Reply:
Yes, you are right. If I understand correctly:
- if first is then also second is because we check both sources -> file system is reliable
- if first is not (due to coincidence of being created at exact second), but second is -> file system is reliable

Reply:
Exactly!

Reply:
Another option is also that the second was read via some other source, like using native platform. So we may still get some low-precision timestamps.